fs-tep-api/src/main/java/com/cgi/eoss/fstep/api/controllers/DataBasketsApiExtension.java

```diff
@@ -0,0 +1,145 @@
+package com.cgi.eoss.fstep.api.controllers;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import org.geojson.Feature;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.rest.webmvc.BasePathAwareController;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import com.cgi.eoss.fstep.catalogue.CatalogueService;
+import com.cgi.eoss.fstep.model.Databasket;
+import com.cgi.eoss.fstep.model.FstepFile;
+import com.cgi.eoss.fstep.persistence.service.DatabasketDataService;
+import com.cgi.eoss.fstep.search.api.SearchResults;
+import com.cgi.eoss.fstep.search.api.SearchResults.Page;
+import lombok.extern.log4j.Log4j2;
+
+@RestController
+@BasePathAwareController
+@RequestMapping("/databaskets")
+@Log4j2
+public class DataBasketsApiExtension {
+
+    private SearchApi searchApi;
+    private CatalogueService catalogueService;
+    private DatabasketDataService databasketDataService;
+    
+    private int maxItemsDatabasketsFromSearch;
+    
+    @Autowired
+    public DataBasketsApiExtension(SearchApi searchApi, CatalogueService catalogueService,
+            DatabasketDataService databasketDataService, @Value("${fstep.api.maxItemsDatabasketsFromSearch:500}") int maxItemsDatabasketsFromSearch) {
+        this.searchApi = searchApi;
+        this.catalogueService = catalogueService;
+        this.databasketDataService = databasketDataService;
+        this.maxItemsDatabasketsFromSearch = maxItemsDatabasketsFromSearch;
+    }
+
+
+    @PostMapping("/{databasketId}/fromSearch")
+    @PreAuthorize("hasAnyRole('CONTENT_AUTHORITY', 'ADMIN') or hasPermission(#databasket, 'write')")
+    ResponseEntity fromSearch(@ModelAttribute("databasketId") Databasket databasket, HttpServletRequest request) {
+        try {
+            LOG.debug("Populating databasket {} with search results", databasket.getId());
+            SearchResults searchResults = searchApi.search(request);
```

Comment:
Maybe instead of this, you could inject the SearchFacade and create the SearchParameters object directly? It would maybe be easier to use this programmatically vs. manipulating the Request with additional parameters to grab extra pages and stuff.

Reply:
I didn't want to replicate the logic that transforms the request into search parameters. I could extract that part into a utility method somewhere in the search api package, call that and use the searchfacade. I'll add a TODO and merge the current version in the meantime.

Reply:
SGTM